File: app/src/main/java/im/tox/antox/callbacks/AntoxOnTypingChangeCallback.java
package im.tox.antox.callbacks;

import android.content.Context;

import im.tox.antox.tox.ToxSingleton;
import im.tox.antox.utils.AntoxFriend;
import im.tox.jtoxcore.callbacks.OnTypingChangeCallback;

public class AntoxOnTypingChangeCallback implements OnTypingChangeCallback<AntoxFriend> {

    private static final String TAG = "OnTypingChangeCallback";
    private Context ctx;

    private ToxSingleton toxSingleton = ToxSingleton.getInstance();

    public AntoxOnTypingChangeCallback(Context ctx) {
        this.ctx = ctx;
    }

    public void execute(AntoxFriend friend, boolean typing) {
        toxSingleton.typingMap.put(friend.getId(),typing);
        toxSingleton.typingSubject.onNext(true);
    }
}

File: app/src/main/scala/im/tox/antox/callbacks/AntoxOnTypingChangeCallback.scala
package im.tox.antox.callbacks

import android.content.Context
import im.tox.antox.tox.{Reactive, ToxSingleton}
import im.tox.tox4j.core.callbacks.FriendTypingCallback

object AntoxOnTypingChangeCallback {

  private val TAG = "OnTypingChangeCallback"
}

class AntoxOnTypingChangeCallback(private var ctx: Context) extends FriendTypingCallback {

  override def friendTyping(friendNumber: Int, isTyping: Boolean): Unit = {
    ToxSingleton.typingMap.put(ToxSingleton.getAntoxFriend(friendNumber).get.getKey, isTyping)
    Reactive.typing.onNext(true)
  }
}
